core/os/input_event.cpp

```diff
@@ -34,8 +34,52 @@
  */
 
 bool InputEvent::operator==(const InputEvent &p_event) const {
+	if (type != p_event.type){
+		return false;
+	}
```

Comment:
not check further if type is not same, like `InputEventKey` != `InputEventMouseButton`.
rest `case` didn't check type is same.